animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ProductModelImpl.java

```diff
@@ -75,6 +84,57 @@ public void authorize(Aspect<Product> aspect, Attribute... attributes) {
         }
     }
 
+    /* (non-Javadoc)
+     * @see com.chiralbehaviors.CoRE.meta.NetworkedModel#authorize(com.chiralbehaviors.CoRE.ExistentialRuleform, com.chiralbehaviors.CoRE.relationship.Relationship, com.chiralbehaviors.CoRE.agency.Agency)
+     */
+    @Override
+    public void authorize(Product ruleform, Relationship relationship,
+                          Agency authorized) {
+        AgencyProduct a = new AgencyProduct(kernel.getCoreAnimationSoftware());
```

Comment:
Wait a minute. Ignoring my style concerns, there's a bug here. Given agency A, Product P, and Relationship R, this code produces 2 AgencyProducts:
A R P
A R.Inverse P

That makes no sense. It should be P R.Inverse A